### ISSUE TITLE
[Bug] Vesktop flatpak spawns an invisible window that cannot be interacted with.

### ISSUE THREAD

**Issue by MythicalNewt** [Vencord/Vesktop#581]
### Discord Account

mythicalnewt

### Operating System

OpenSUSE Tumbleweed

### Linux Only ~ Desktop Environment

KDE Plasma 6 on wayland

### What happens when the bug or crash occurs?

When vesktop is launched using krunner the loading vesktop window spawns, followed by an invisible fullscreen window that cannot be interacted with. 

Closing this window and clicking on the vesktop tray icon works around this issue.

 I have included a video to illustrate the issue better:

https://github.com/Vencord/Vesktop/assets/89124240/17d93372-f0cf-4e76-afcb-08ca6172222d

### What is the expected behaviour?

The window highlighted by the taskbar should be interactable.

### How do you recreate this bug or crash?

1. Launch vesktop flatpak on OpenSUSE Tumbleweed

### Debug Logs

```
$ flatpak run dev.vencord.Vesktop 

Passing the following arguments to Electron: --ozone-platform-hint=auto
[3:0506/073731.058282:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[3:0506/073731.405754:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[3:0506/073731.405814:ERROR:bus.cc(407)] Failed to connect to the bus: Failed to connect to socket /run/dbus/system_bus_socket: No such file or directory
[arRPC > ipc] listening at /run/user/1000/discord-ipc-0
[arRPC > websocket] listening on 6463
[arRPC > process] started
[53:0506/073731.754891:ERROR:gbm_pixmap_wayland.cc(82)] Cannot create bo with format= RGBA_8888 and usage=SCANOUT
[53:0506/073731.755028:ERROR:gbm_pixmap_wayland.cc(82)] Cannot create bo with format= RGBA_8888 and usage=GPU_READ
[53:0506/073731.755090:ERROR:shared_image_factory.cc(984)] CreateSharedImage: could not create backing.
[53:0506/073731.755180:ERROR:shared_image_factory.cc(767)] DestroySharedImage: Could not find shared image mailbox
[53:0506/073731.755356:ERROR:gpu_service_impl.cc(1154)] Exiting GPU process because some drivers can't recover from errors. GPU process will restart shortly.
[3:0506/073731.763392:ERROR:gpu_process_host.cc(989)] GPU process exited unexpectedly: exit_code=8704
[148:0506/073731.865145:ERROR:gbm_pixmap_wayland.cc(82)] Cannot create bo with format= RGBA_8888 and usage=SCANOUT
[148:0506/073731.865212:ERROR:gbm_pixmap_wayland.cc(82)] Cannot create bo with format= RGBA_8888 and usage=GPU_READ
[148:0506/073731.865251:ERROR:shared_image_factory.cc(984)] CreateSharedImage: could not create backing.
[148:0506/073731.865311:ERROR:shared_image_factory.cc(767)] DestroySharedImage: Could not find shared image mailbox
[148:0506/073731.865429:ERROR:gpu_service_impl.cc(1154)] Exiting GPU process because some drivers can't recover from errors. GPU process will restart shortly.
[3:0506/073731.871481:ERROR:gpu_process_host.cc(989)] GPU process exited unexpectedly: exit_code=8704
[3:0506/073731.973099:ERROR:interface_endpoint_client.cc(707)] Message 6 rejected by interface blink.mojom.WidgetHost
[3:0506/073731.973127:ERROR:interface_endpoint_client.cc(707)] Message 7 rejected by interface blink.mojom.WidgetHost
[198:0506/073731.974960:ERROR:gbm_pixmap_wayland.cc(82)] Cannot create bo with format= RGBA_8888 and usage=SCANOUT
[198:0506/073731.975030:ERROR:gbm_pixmap_wayland.cc(82)] Cannot create bo with format= RGBA_8888 and usage=GPU_READ
[198:0506/073731.975101:ERROR:shared_image_factory.cc(984)] CreateSharedImage: could not create backing.
[198:0506/073731.975172:ERROR:shared_image_factory.cc(767)] DestroySharedImage: Could not find shared image mailbox
[198:0506/073731.975276:ERROR:gpu_service_impl.cc(1154)] Exiting GPU process because some drivers can't recover from errors. GPU process will restart shortly.
[3:0506/073731.981756:ERROR:gpu_process_host.cc(989)] GPU process exited unexpectedly: exit_code=8704
[118:0506/073732.061435:ERROR:command_buffer_proxy_impl.cc(131)] ContextResult::kTransientFailure: Failed to send GpuControl.CreateCommandBuffer.

(vesktop.bin:3): libnotify-WARNING **: 07:37:48.154: Running in confined mode, using Portal notifications. Some features and hints won't be supported
```

A bunch of gpu acceleration errors seem to be shown I have included output of vainfo below:

```
$ vainfo   

Trying display: wayland
libva info: VA-API version 1.21.0
libva info: Trying to open /usr/lib64/dri/radeonsi_drv_video.so
libva info: Found init function __vaDriverInit_1_21
libva info: va_openDriver() returns 0
vainfo: VA-API version: 1.21 (libva 2.20.1)
vainfo: Driver version: Mesa Gallium driver 24.0.5 for AMD Radeon RX 590 Series (radeonsi, polaris10, LLVM 18.1.4, DRM 3.57, 6.8.8-lqx1-1-liquorix)
vainfo: Supported profile and entrypoints
      VAProfileMPEG2Simple            :	VAEntrypointVLD
      VAProfileMPEG2Main              :	VAEntrypointVLD
      VAProfileVC1Simple              :	VAEntrypointVLD
      VAProfileVC1Main                :	VAEntrypointVLD
      VAProfileVC1Advanced            :	VAEntrypointVLD
      VAProfileH264ConstrainedBaseline:	VAEntrypointVLD
      VAProfileH264ConstrainedBaseline:	VAEntrypointEncSlice
      VAProfileH264Main               :	VAEntrypointVLD
      VAProfileH264Main               :	VAEntrypointEncSlice
      VAProfileH264High               :	VAEntrypointVLD
      VAProfileH264High               :	VAEntrypointEncSlice
      VAProfileHEVCMain               :	VAEntrypointVLD
      VAProfileHEVCMain               :	VAEntrypointEncSlice
      VAProfileHEVCMain10             :	VAEntrypointVLD
      VAProfileJPEGBaseline           :	VAEntrypointVLD
      VAProfileNone                   :	VAEntrypointVideoProc
```

### Request Agreement

- [X] I have searched the existing issues and found no similar issue
- [X] I am using the latest Vesktop and Vencord versions
- [X] This issue occurs on an official release (not just the AUR or Nix packages)

**Comment by MythicalNewt** [comment]
This bug has been happening since 1.5.1 current version showcased in video is 1.5.2

```$ flatpak info dev.vencord.Vesktop       

Vesktop - Snappier Discord app with Vencord

          ID: dev.vencord.Vesktop
         Ref: app/dev.vencord.Vesktop/x86_64/stable
        Arch: x86_64
      Branch: stable
     Version: 1.5.2
     License: GPL-3.0
      Origin: flathub
  Collection: org.flathub.Stable
Installation: system
   Installed: 281.6 MB
     Runtime: org.freedesktop.Platform/x86_64/23.08
         Sdk: org.freedesktop.Sdk/x86_64/23.08

      Commit: 5b639112007e4d324118ef7adf930de3b7b074ea1201957e815911c5d7b51778
      Parent: b03a262a4a03e08167e04bc0a65ae09ff6f520882c1f4c40da02ff0895911899
     Subject: Update to version 1.5.2 (#17) (84aa7cb8)
        Date: 2024-05-02 18:21:41 +0000```

**Comment by Vendicated** [comment]
seems like a gpu issue. have you tried disabling hardware acceleration or running with --disable-gpu?

**Comment by MythicalNewt** [comment]
Yes it seems to be a GPU issue using that flag fixes it. Any idea whats wrong with gpu acceleration, although I understand its out of scope of this issue, it used to work fine before then.

**Comment by Vendicated** [comment]
sadly not, although we've had many reports of kde users so it might be a kde issue

if the disable gpu flag works for you and doesn't lag, should be fine? although you'll have higher cpu usage and not a good time streaming your screen

**Comment by MythicalNewt** [comment]
i see, yea the streaming could be a problem, perhaps i could give the appimage a try
i did try enabling the discrete gpu flatpak permissions through kde settings that doesnt seem to have effected it
ill get back to you on how it went in sometime

**Comment by MythicalNewt** [comment]
The appimage seems to be working perfectly fine, looks like its a flatpak permissions/ shared libraries issue on kde systems.

**Comment by MetriC-DT** [comment]
I don't seem to have a problem on systems with iGPUs (e.g. my laptop). Unfortunately, my desktop doesn't have an iGPU and it seems to run into the GPU bug...both are AMD.

**Comment by MetriC-DT** [comment]
unfortunately no dice (I had never touched that file - this is under the flatpak config directory). Tried deleting that file, but issue still persists

**Comment by MetriC-DT** [comment]
Hi, so the issue seems to be resolved by using electron version 31.1.0. Hardware acceleration still fails on wayland, but at least discord can be viewed instead of just an invisible window...
The GPU errors are all still there using these flags:
```
--ozone-platform-hint=auto" "--enable-webrtc-pipewire-capturer" "--enable-features=WaylandWindowDecorations" "--enable-features=UseOzonePlatform" "--ozone-platform=wayland" "--use-gl=angle" "--use-angle=gl" "--enable-features=VaapiVideoDecodeLinuxGL,VaapiVideoEncoder
```

The error messages:
```
[22080:0628/111252.329637:ERROR:gbm_pixmap_wayland.cc(82)] Cannot create bo with format= RGBA_8888 and usage=SCANOUT
[22080:0628/111252.329744:ERROR:gbm_pixmap_wayland.cc(82)] Cannot create bo with format= RGBA_8888 and usage=GPU_READ
[22080:0628/111252.329801:ERROR:shared_image_factory.cc(1015)] CreateSharedImage: could not create backing.
[22080:0628/111252.329888:ERROR:shared_image_factory.cc(764)] DestroySharedImage: Could not find shared image mailbox
[22080:0628/111252.330011:ERROR:gpu_service_impl.cc(1125)] Exiting GPU process because some drivers can't recover from errors. GPU process will restart shortly.
[22036:0628/111252.345202:ERROR:gpu_process_host.cc(1002)] GPU process exited unexpectedly: exit_code=8704
[22036:0628/111252.481163:ERROR:interface_endpoint_client.cc(722)] Message 6 rejected by interface blink.mojom.WidgetHost
[22036:0628/111252.481181:ERROR:interface_endpoint_client.cc(722)] Message 7 rejected by interface blink.mojom.WidgetHost
[22161:0628/111252.495372:ERROR:gbm_pixmap_wayland.cc(82)] Cannot create bo with format= RGBA_8888 and usage=SCANOUT
[22161:0628/111252.495469:ERROR:gbm_pixmap_wayland.cc(82)] Cannot create bo with format= RGBA_8888 and usage=GPU_READ
[22161:0628/111252.495507:ERROR:shared_image_factory.cc(1015)] CreateSharedImage: could not create backing.
[22161:0628/111252.495551:ERROR:shared_image_factory.cc(764)] DestroySharedImage: Could not find shared image mailbox
[22161:0628/111252.495641:ERROR:gpu_service_impl.cc(1125)] Exiting GPU process because some drivers can't recover from errors. GPU process will restart shortly.
[22114:0628/111252.502315:ERROR:command_buffer_proxy_impl.cc(325)] GPU state invalid after WaitForGetOffsetInRange.
[22114:0628/111252.502720:ERROR:command_buffer_proxy_impl.cc(132)] ContextResult::kTransientFailure: Failed to send GpuControl.CreateCommandBuffer.
[22036:0628/111252.513166:ERROR:gpu_process_host.cc(1002)] GPU process exited unexpectedly: exit_code=8704
[22222:0628/111252.647225:ERROR:gbm_pixmap_wayland.cc(82)] Cannot create bo with format= RGBA_8888 and usage=SCANOUT
[22222:0628/111252.647297:ERROR:gbm_pixmap_wayland.cc(82)] Cannot create bo with format= RGBA_8888 and usage=GPU_READ
[22222:0628/111252.647381:ERROR:shared_image_factory.cc(1015)] CreateSharedImage: could not create backing.
[22222:0628/111252.647439:ERROR:shared_image_factory.cc(764)] DestroySharedImage: Could not find shared image mailbox
[22222:0628/111252.647529:ERROR:gpu_service_impl.cc(1125)] Exiting GPU process because some drivers can't recover from errors. GPU process will restart shortly.
[22036:0628/111252.655969:ERROR:gpu_process_host.cc(1002)] GPU process exited unexpectedly: exit_code=8704
[22114:0628/111252.769308:ERROR:command_buffer_proxy_impl.cc(132)] ContextResult::kTransientFailure: Failed to send GpuControl.CreateCommandBuffer.
```
